### PR TITLE
commit changes for main branch

### PR DESCRIPTION
**Pull request by mafatullayev**

### CODE REVIEW

File: src/main/java/com/example/msuser/mapper/UserMapper.java [hunk 1]
```diff
@@ -39,7 +39,6 @@ public UserEntity buildUpdateUserEntity(UpdateUserRequest updateUserRequest, Lon
                 .mail(updateUserRequest.getMail())
                 .photo(updateUserRequest.getPhoto())
                 .password(updateUserRequest.getPassword())
```
Comment (ilkin-aliyev-uz) [inline src/main/java/com/example/msuser/mapper/UserMapper.java:41]: .type(updateUserRequest.getType)

File: src/main/java/com/example/msuser/dao/entity/UserEntity.java [hunk 2]
```diff
@@ -17,19 +32,28 @@
...
     private String name;
     private String surname;
     private String password;
     private String mail;
-    private String userType;
-    private String photo;
 
     @Enumerated(STRING)
-    private UserStatus status;
+    @Column(name = "type")
```
Comment (ilkin-aliyev-uz) [inline src/main/java/com/example/msuser/dao/entity/UserEntity.java]: remove @Column

File: src/main/java/com/example/msuser/exception/ErrorHandlerException.java [hunk 3]
```diff
@@ -13,26 +12,19 @@
 
 @Slf4j
 @RestControllerAdvice
-public class ErrorHandler {
+public class ErrorHandlerException {
```
Comment (ilkin-aliyev-uz) [inline src/main/java/com/example/msuser/exception/ErrorHandlerException.java]: revert to ErrorHandler

File: src/main/java/com/example/msuser/exception/ErrorHandlerException.java [hunk 4]
```diff
@@ -13,26 +12,19 @@
...
 
     @ExceptionHandler(Exception.class)
     @ResponseStatus(INTERNAL_SERVER_ERROR)
     public ExceptionResponse handle(Exception ex) {
         log.error("Exception: ", ex);
-        return new ExceptionResponse();
+        return new ExceptionResponse("Internal Server Error");
```
Comment (ilkin-aliyev-uz) [inline src/main/java/com/example/msuser/exception/ErrorHandlerException.java]: move this to constant and dont use Internal Server Error. You can use for example, Unexpected error occurred

File: src/main/java/com/example/msuser/mapper/UserMapper.java [hunk 5]
```diff
@@ -24,9 +22,11 @@ public UserEntity buildUserEntity(CreateUserRequest userRequest) {
     public UserResponse buildUserResponse(UserEntity userEntity) {
         return UserResponse
                 .builder()
-                .id(userEntity.getId())
                 .surname(userEntity.getSurname())
                 .name(userEntity.getName())
+                .mail(userEntity.getMail())
+                .userType(userEntity.getType())
+                .birthDate(userEntity.getBirthDate())
```
Comment (ilkin-aliyev-uz) [inline src/main/java/com/example/msuser/mapper/UserMapper.java:30]: add type to buildUserEntity  and buildUpdateUserEntity

File: src/test/groovy/com/example/msuser/UserServiceHandlerTest.groovy [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+    def "TestSignUp"() {
+        given:
+        def request = random.nextObject(CreateUserRequest)
+        def entity = USER_MAPPER.buildUserEntity(request)
+
+        when:
+        userService.signUp(request)
+
+        then:
+        1 * request.setPassword(request)
```
Comment (ilkin-aliyev-uz) [inline src/test/groovy/com/example/msuser/UserServiceHandlerTest.groovy]: use calls count only for beans

File: src/test/groovy/com/example/msuser/UserServiceHandlerTest.groovy [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+        1 * userRepository.save(entity)
+    }
+
+    def "TestSignIn"() {
```
Comment (ilkin-aliyev-uz) [inline src/test/groovy/com/example/msuser/UserServiceHandlerTest.groovy]: also add test for success case

File: src/test/groovy/com/example/msuser/UserServiceHandlerTest.groovy [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+        NotFoundException ex = thrown()
+        ex.message == "User not found!"
+    }
+
+    def "TestUpdateUser error user can't update"() {
+//        give:
+//        when:
+//        then:
+    }
+
```
Comment (ilkin-aliyev-uz) [inline src/test/groovy/com/example/msuser/UserServiceHandlerTest.groovy:121]: def "TestUpdateUser success"() {
        given:
        def id = random.nextLong()
        def request = random.nextObject(UpdateUserRequest)
        def entity = random.nextObject(UserEntity)
        def savedEntity = (-- based on entity object)
        def expected = USER_MAPPER.buildUpdateUserEntity(id, request)


        when:
        def response = userService.updateUser(id, request)

        then:
        1 * userRepository.findById(id) >> Optional.of(entity)
        1 * userRepository.save(savedEntity)  
    }

File: src/main/java/com/example/msuser/dao/entity/UserEntity.java (hunk 2 above)
Comment (mafatullayev) [reply src/main/java/com/example/msuser/dao/entity/UserEntity.java]: DONE

**Pull request closed without merging**